app/uk/gov/hmrc/gform/sharedmodel/form/SubmissionData.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gform.sharedmodel.form
+
+import uk.gov.hmrc.gform.sharedmodel.VariableKey
+import play.api.libs.json.Json.JsValueWrapper
+import play.api.libs.json._
+
+case class SubmissionData(pdfData: String, variables: Variables)
+
+object SubmissionData {
+  implicit val format: OFormat[SubmissionData] = Json.format[SubmissionData]
+}
+
+case class Variables(value: Map[VariableKey, String])
+
+object Variables {
+
+  implicit val writes = new Writes[Map[VariableKey, String]] {
+    def writes(variables: Map[VariableKey, String]): JsValue =
+      Json.obj(variables.map {
+        case (k, v) =>
+          val tuple: (String, JsValueWrapper) = k -> v
+          tuple
+      }.toSeq: _*)
+  }
+
+  implicit val reads = new Reads[Map[VariableKey, String]] {
+    def reads(jsValue: JsValue): JsResult[Map[VariableKey, String]] =
+      jsValue match {
+        case JsObject(v) =>
+          JsSuccess(v.toMap.map {
+            case (k, v) => k -> v.as[String]
+          })
+        case _ => JsError(s"Unable to read $jsValue")
+      }
+  }
+
+  implicit val formatM: Format[Map[VariableKey, String]] = Format(reads, writes)
```

Comment:
Same question as in: https://github.com/hmrc/gform-frontend/pull/654/files#r261205307

Reply:
done